optimism/op-node/rollup/engine/preconf.go

```diff
@@ -148,6 +153,16 @@ func (s *PreconfState) putSeal(sSeal *eth.SignedSeal) {
 	}
 }
 
+// Checks if there's envs blocked because of gaps and sends them over.
+func (s *PreconfState) putL2Block(block *eth.L2BlockRef) {
+	s.sentL2Blocks[block.Number] = true
```

Comment:
Can't this always-growing hashmap be a problem after some time?

Reply:
Yes, the same with all the rest. I'll add pruning for all in a separate thread.